File: java/program/Mekanik.java
public class Mekanik extends Manusia {
    private String spesialisasi;
    private int pengalamanTahun;

    public Mekanik(String nama, String alamat, String spesialisasi, int pengalamanTahun) {
        super(nama, alamat);
        this.spesialisasi = spesialisasi;
        this.pengalamanTahun = pengalamanTahun;
    }

    public String getSpesialisasi() {
        return spesialisasi;
    }

    public void setSpesialisasi(String spesialisasi) {
        this.spesialisasi = spesialisasi;
    }

    public int getPengalamanTahun() {
        return pengalamanTahun;
    }

    public void setPengalamanTahun(int pengalamanTahun) {
        this.pengalamanTahun = pengalamanTahun;
    }
}

File: cpp/program/Mekanik.cpp
#include "Mekanik.h"

Mekanik::Mekanik(string nama, string alamat, string spesialisasi, int pengalamanTahun)
    : Manusia(nama, alamat), _spesialisasi(spesialisasi), _pengalamanTahun(pengalamanTahun) {}

string Mekanik::getSpesialisasi() { 
    return _spesialisasi; 
}
